### PR TITLE
fix(curriculum): updates `body-parser` readme with better explanation for `extended` config 

### LINKED ISSUE #42682: Clarification for express body-parser exercise
**Affected page**
Add a link to the coding challenge with the problem.
Page for exercise: https://www.freecodecamp.org/learn/apis-and-microservices/basic-node-and-express/use-body-parser-to-parse-post-requests

**Your code**
Copy and paste the code from the editor that you used in between the back-ticks below:

```
// external packages
const express = require('express');
const app = express();
const bodyParser = require('body-parser')

const userPostData = (req,res) => console.log(req.body) 
const bp = bodyParser.urlencoded({extended: false})
app.route("/name").get(userGetData).post(bp,userPostData)
// error at printing object
```

**Expected behavior**
Add a clear and concise description of what you expected to happen.

If we add `extended:false` not only we enable standard query string parsing, but also disable the ability to print the object, this is because the querystring object does not inherit from Object, for security reasons, but can confuse any beginner (I guess).

Can be solved by using `extended:true`, this way, the expected behavior will happen (object is logged out).

**Additional context**
IMHO the exercise should clarify this (although users will found answers on SO if they look it up).

### PR DESCRIPTION
**Pull request by anirudh-modi**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #42682

### CODE REVIEW

**Comment by anirudh-modi** [comment]
Hi @raisedadead @ojeytonwilliams

- I could not found any tests for the same.. have I missed anything?
- Is there any standard process for how changes for other languages should be done? (I tried referring to the steps mentioned in https://contribute.freecodecamp.org/#/how-to-open-a-pull-request for language, but perhaps the link is broken?)

Any guidance here would be helpful..

**Comment by ojeytonwilliams** [comment]
Hi @anirudh-modi, thanks for the PR.  Regarding tests, I'm not 100% what the process for adding new backend tests is @ShaunSHamilton or @scissorsneedfoodtoo might know more. It's probably fine, though, since the main goal is just to get the user to use `body-parser`.

**Comment by ojeytonwilliams** [comment]
Don't worry about the translations, though.  If you're interested in the process, take a look at https://contribute.freecodecamp.org/#/how-to-translate-files, but, TL;DR, we don't submit PRs for translations any more - just the original English text.

File: curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/use-body-parser-to-parse-post-requests.md [hunk 1]
```diff
@@ -26,9 +26,11 @@ As you can see, the body is encoded like the query string. This is the default f
...
 
-**Note:** `extended=false` is a configuration option that tells the parser to use the classic encoding. When using it, values can be only strings or arrays. The extended version allows more data flexibility, but it is outmatched by JSON.
+**Note:** `extended` is a configuration option that tells `body-parser` which parsing needs to be used. When `extended=false` it uses the classic encoding `querystring` library. When `extended=true` it uses `qs` library for parsing. 
+
+When using `extended=false`, values can be only strings or arrays. The object returned when using `querystring` does not prototypically inherit from the default Javascript `Object`, which means functions like `hasOwnProperty`, `toString` will not be available. The extended version allows more data flexibility, but it is outmatched by JSON.
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/use-body-parser-to-parse-post-requests.md:33]: With the addition of this line, I do not think it is necessary to change the encoding to use the extended parser.

**Comment by anirudh-modi** [comment]
Hi thanks, for the feedback let me resolve the same

File: curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/use-body-parser-to-parse-post-requests.md [hunk 2]
```diff
@@ -26,9 +26,11 @@ As you can see, the body is encoded like the query string. This is the default f
 
 # --instructions--
 
-Install the `body-parser` module in your `package.json`. Then, `require` it at the top of the file. Store it in a variable named `bodyParser`. The middleware to handle urlencoded data is returned by `bodyParser.urlencoded({extended: false})`. Pass to `app.use()` the function returned by the previous method call. As usual, the middleware must be mounted before all the routes which need it.
+Install the `body-parser` module in your `package.json`. Then, `require` it at the top of the file. Store it in a variable named `bodyParser`. The middleware to handle urlencoded data is returned by `bodyParser.urlencoded({extended: true})`. Pass to `app.use()` the function returned by the previous method call. As usual, the middleware must be mounted before all the routes which need it.
```
Comment (alirezaghey) [inline curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/use-body-parser-to-parse-post-requests.md:29]:
```suggestion
Install the `body-parser` module in your `package.json`. Then, `require` it at the top of the file. Store it in a variable named `bodyParser`. The middleware to handle urlencoded data is returned by `bodyParser.urlencoded({extended: true})`. Pass the function returned by the previous method call to `app.use()`. As usual, the middleware must be mounted before all the routes that depend on it.
```

Comment (anirudh-modi) [reply curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/use-body-parser-to-parse-post-requests.md]: Hi @alirezaghey I have commited the suggestion you have asked for.

File: curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/use-body-parser-to-parse-post-requests.md (hunk 1 above)
Comment (anirudh-modi) [reply curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/use-body-parser-to-parse-post-requests.md:33]: Hi @ShaunSHamilton I have reverted the encoding to `false` as it was there originally

**Review by ShaunSHamilton — DISMISSED** [review]
This LGTM 👍

**Comment by anirudh-modi** [comment]
I hope I do not have to do anything else @alirezaghey @ShaunSHamilton for this PR..

**Review by alirezaghey — DISMISSED** [review]
LGTM :+1:

**Comment by anirudh-modi** [comment]
Awesome!! But weirdly the merge is still blocking..

**Comment by alirezaghey** [comment]
> Awesome!! But weirdly the merge is still blocking..

Approval without write access doesn't count toward mergeability, which makes sense. Otherwise, any combination of 3 distinct github accounts could merge changes to the codebase.

File: curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/use-body-parser-to-parse-post-requests.md [hunk 3]
```diff
@@ -26,9 +26,11 @@ As you can see, the body is encoded like the query string. This is the default f
...
 
-**Note:** `extended=false` is a configuration option that tells the parser to use the classic encoding. When using it, values can be only strings or arrays. The extended version allows more data flexibility, but it is outmatched by JSON.
+**Note:** `extended` is a configuration option that tells `body-parser` which parsing needs to be used. When `extended=false` it uses the classic encoding `querystring` library. When `extended=true` it uses `qs` library for parsing. 
+
+When using `extended=false`, values can be only strings or arrays. The object returned when using `querystring` does not prototypically inherit from the default Javascript `Object`, which means functions like `hasOwnProperty`, `toString` will not be available. The extended version allows more data flexibility, but it is outmatched by JSON.
```
Comment (huyenltnguyen) [inline curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/use-body-parser-to-parse-post-requests.md:33]: 😛 

```suggestion
When using `extended=false`, values can be only strings or arrays. The object returned when using `querystring` does not prototypically inherit from the default JavaScript `Object`, which means functions like `hasOwnProperty`, `toString` will not be available. The extended version allows more data flexibility, but it is outmatched by JSON.
```

**Review by huyenltnguyen — CHANGES_REQUESTED** [review]
There is just a small typo. LGTM otherwise 👍

**Comment by alirezaghey** [comment]
hah, nice catch :tada:

**Comment by anirudh-modi** [comment]
Thanks for pointing the same 😊

Committed the suggestion.

**Review by huyenltnguyen — APPROVED** [review]
LGTM!

@anirudh-modi Thanks for being patient with us and congratulations on your first contribution to the repo 🎉

**Comment by anirudh-modi** [comment]
No problem at all!! 🙂

It was my pleasure that I could contribute to freeCodeCamp 🙂

**Review by ShaunSHamilton — APPROVED** [review]

**Pull request merged**